[EPMEDU-1089] Create About page

aklepets opened the pull request:
[EPMEDU-1089] Create About page
Facebook and Instagram buttons open corresponding app if any. else open links in browser
Linkedin button opens browser. 
A web link leads to Linkedin for now. To be updated when the correct link is provided.

ArtemKleinschmidt committed:
[EPMEDU-968]: Open the feeding point details from the Favorites page (#109)

* [EPMEDU-1068] Open the feeding point details from the Favorites page

Moved HomeBottomSheet to foundation
Moved FeedingPointSheetContent and dependencies to shared
Added feeding details dialog to the favorites screen without modifications

* Make feed button unavailable is an animal is fed
Added "go to map" link
Extracted alpha calculation logic to make it shared

egorikftp committed:
Increment build version

Fileek committed:
[EPMEDU-1090]: Rename Help page to FAQ (#114)

egorikftp committed:
Increment build version

aklepets committed:
[EPMEDU-1089] Create About page

Fileek commented on library/resources/src/main/res/drawable/icon_fb.xml:
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
We have `ic_facebook` already

Fileek commented on library/resources/src/main/res/drawable/icon_inst.xml:
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Let's use the same format for all of our drawable names: `ic_instagram`

Fileek commented on library/resources/src/main/res/drawable/icon_link.xml:
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
same as above

Fileek commented on library/resources/src/main/res/drawable/icon_web.xml:
```diff
@@ -0,0 +1,54 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
same as above

Fileek commented on library/foundation/src/main/java/com/epmedu/animeal/foundation/button/AnimealSocialButton.kt:
```diff
@@ -0,0 +1,77 @@
+
+@AnimealPreview
+@Composable
+private fun FbAnimealSocialButtonPreview() {
```
Please put all previews together in one preview function and use column or row

Fileek commented on feature/tabsflow/moreflow/about/build.gradle.kts:
```diff
@@ -18,4 +18,5 @@ dependencies {
     implementation(libs.compose.ui)
     implementation(libs.compose.ui.preview)
     debugImplementation(libs.compose.ui.tooling)
+    implementation(libs.compose.material)
```
Sort by alphabetical order

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreenUI.kt:
```diff
@@ -1,20 +1,93 @@
-        onBack = onBack,
-    )
+    Column(
+        modifier = modifier.fillMaxSize(),
```
Add `.statusBarsPadding()`

Fileek commented on library/foundation/src/main/java/com/epmedu/animeal/foundation/button/AnimealSocialButton.kt:
```diff
@@ -0,0 +1,77 @@
+    iconResource: Int
+) {
+    FloatingActionButton(
+        modifier = modifier.size(62.dp),
```
Can you add gradient to match the design? It's similiar to `BottomAppBarFab`

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreenUI.kt:
```diff
@@ -1,20 +1,93 @@
+        )
+        Text(
+            modifier = Modifier.padding(start = 32.dp, end = 32.dp, top = 12.dp, bottom = 12.dp),
+            text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, " +
```
There is `LoremIpsum` class for that :)

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreenUI.kt:
```diff
@@ -1,20 +1,93 @@
+        )
+        Text(
+            modifier = Modifier.padding(start = 32.dp, end = 32.dp, top = 12.dp, bottom = 18.dp),
+            text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, " +
```
Same as above

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreenUI.kt:
```diff
@@ -1,20 +1,93 @@
+            navigationIcon = { BackButton(onClick = onBack) }
+        )
+        Image(
+            modifier = Modifier.padding(start = 64.dp, end = 64.dp, top = 18.dp, bottom = 18.dp)
```
Can be replaced with `horizontal` and `vertical` arguments

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreenUI.kt:
```diff
@@ -1,20 +1,93 @@
-        title = stringResource(id = R.string.page_about),
-        onBack = onBack,
-    )
+    Column(
```
Use `Scaffold` as a root with `topBar` so that inside it you can apply horizontal paddings on the whole column, not to each element

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreen.kt:
```diff
@@ -9,10 +14,63 @@ import com.epmedu.animeal.navigation.navigator.LocalNavigator
     val navigator = LocalNavigator.currentOrThrow
+    val context = LocalContext.current
+    val facebookWebIntent = remember {
+        Intent(Intent.ACTION_VIEW, Uri.parse("https://www.facebook.com/animalprojectgeorgia/"))
```
Please extract all links to constants in `library/common/constants/Links`

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreen.kt:
```diff
@@ -9,10 +14,63 @@ import com.epmedu.animeal.navigation.navigator.LocalNavigator
+            }
+        },
+        onSocialInstagramClick = {
+            try {
```
Don't repeat yourself. Extract the try-catch block to extension function `Context.startAppOrBrowser(
    appIntent: Intent,
    webIntent: Intent
)` in `library/extensions/Context`

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreen.kt:
```diff
@@ -9,10 +14,63 @@ import com.epmedu.animeal.navigation.navigator.LocalNavigator
 fun AboutScreen() {
     val navigator = LocalNavigator.currentOrThrow
+    val context = LocalContext.current
+    val facebookWebIntent = remember {
```
What do you think about extracting all these intents to extension functions like `Context.openFacebook()`?

Fileek committed:
[EPMEDU-1173]: Align code confirmation logic with backend to allow multiple entry attempts (#116)

Co-authored-by: Uladzislau Balatayeu <uladzislau_balatayeu@epam.com>

egorikftp committed:
Increment build version

Fileek committed:
[EPMEDU-1168]: Phone number is showing incorrectly in profile info (#117)

Prefix and phone number are splitted to prevent this kind of issues and support further prefix changing implementation

Co-authored-by: Uladzislau Balatayeu <uladzislau_balatayeu@epam.com>

egorikftp committed:
Increment build version

Makentoshe committed:
[EPMEDU-867]: Show user on map if Gps enabled (#70)

Andrii-Chornyi committed:
[EPMEDU-1078] Add Immutable Collections Library (#110)

* [EPMEDU-1078] Add Immutable Collections Library

* [EPMEDU-1078] replaced states' kotlin lists on ImmutableList

* [EPMEDU-1078]: updated FavouritesState after it was moved with replacing kotlin list on immutable list

egorikftp committed:
Increment build version

carlossantiagomenendez commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreenUI.kt:
```diff
@@ -1,20 +1,93 @@
+    onSocialFacebookClick: () -> Unit,
+    onSocialInstagramClick: () -> Unit,
+    onSocialLinkedinClick: () -> Unit,
+    onSocialWebClick: () -> Unit,
```
Hi! What do you think about only one listener that has an argument specifying the social media type? Something like: 
`OnSocialClick: (type: SocialMedia) -> Unit` and then having a sealed interface containing objects representing each social media. It's just a suggestion to reduce a little the function arguments.

likaGlonti committed:
[EPMEDU-1131]: Pressing on particular place on map opens feeding screen (#119)

* [EPMEDU-1131]: Pressing on particular place on map opens feeding screen

egorikftp committed:
Increment build version

ArtemKleinschmidt committed:
[EPMEDU-1086]: Create page "Donate" (#118)

* [EPMEDU-1086]: Create page "Donate"

* Added a repository with fake data
Fixed UI in dark mode
Minor refactoring

egorikftp committed:
Increment build version

Fileek committed:
[EPMEDU-1150]: Create UI for FAQ page (#120)

* [EPMEDU-1150]: Create UI for FAQ page

Co-authored-by: Uladzislau Balatayeu <uladzislau_balatayeu@epam.com>

egorikftp committed:
Increment build version

aklepets committed:
fix PR review comments

aklepets committed:
Merge branch 'develop' into feature/EPMEDU-1089_Create_about_page

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreenUI.kt:
```diff
@@ -1,20 +1,107 @@
+    Scaffold(
+        modifier = Modifier
+            .fillMaxSize()
+            .imePadding(),
```
Redundant padding

Fileek replied:
Content should be scrollable, check usecases in your ticket. 
But `Scaffold` does not support this, so you can replace it with `Column` and add `.verticalScroll(rememberScrollState())` to its modifier.

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreenUI.kt:
```diff
@@ -1,20 +1,107 @@
+    ) {
+        Column(
+            modifier = Modifier
+                .statusBarsPadding()
```
Here the padding is not applied to topbar, please use it on the root element or topbar

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreenUI.kt:
```diff
@@ -1,20 +1,107 @@
 
+import android.annotation.SuppressLint
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.*
```
Don't use wildcards

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreen.kt:
```diff
@@ -1,18 +1,35 @@
 import androidx.compose.runtime.Composable
-import com.epmedu.animeal.extensions.currentOrThrow
+import androidx.compose.ui.platform.LocalContext
+import com.epmedu.animeal.extensions.*
```
Same

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreenUI.kt:
```diff
@@ -1,20 +1,107 @@
+    onSocialLinkedinClick: () -> Unit,
+    onSocialWebClick: () -> Unit,
+    text: String,
+    text2: String,
```
The text will be hardcoded in our resources (check jira ticket), there is no need to pass it

Fileek commented on library/extensions/src/main/java/com/epmedu/animeal/extensions/Context.kt:
```diff
@@ -1,5 +1,6 @@
 package com.epmedu.animeal.extensions
 
+import android.content.*
```
Same as above

Fileek reviewed (changes requested):
Fix detekt please

aklepets committed:
fix PR review comments

aklepets committed:
add scroll to Column on About page

aklepets committed:
fix layout About page

aklepets committed:
buttons listener

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreenUI.kt:
```diff
@@ -79,23 +76,17 @@ internal fun AboutScreenUI(
                 lineHeight = 20.sp,
             )
+            Spacer(modifier = Modifier.weight(1f))
+            SocialButtonsRow(
```
These two should be in the outer column

Fileek commented on feature/tabsflow/moreflow/about/src/main/java/com/epmedu/animeal/tabs/more/about/AboutScreenUI.kt:
```diff
@@ -1,20 +1,117 @@
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(bottom = 32.dp, top = 8.dp)
```
Let's use vertical padding instead (32 dp with horizontal)

aklepets committed:
fix pr comment

aklepets committed:
fix padding
